In file included from linuxPci.c:211:
pci.h:454: error: parse error before "pci_power_t"
pci.h:454: warning: type defaults to `int' in declaration of `pci_power_t'
pci.h:454: warning: data definition has no type or storage class

--- X11-6.8.2/xc/programs/Xserver/hw/xfree86/os-support/bus/linuxPci.c.orig	2004-12-15 05:48:20.000000000 +0000
+++ X11-6.8.2/xc/programs/Xserver/hw/xfree86/os-support/bus/linuxPci.c	2005-04-01 11:20:25.678791042 +0000
@@ -45,6 +45,7 @@
  * SOFTWARE.
  */
 
+#include <linux/pci.h>
 #include <stdio.h>
 #include "compiler.h"
 #include "xf86.h"
@@ -208,8 +209,6 @@
  *         master aborts are avoided during PCI scans).
  */
 
-#include <linux/pci.h>
-
 #ifndef PCIIOC_BASE		/* Ioctls for /proc/bus/pci/X/Y nodes. */
 #define PCIIOC_BASE		('P' << 24 | 'C' << 16 | 'I' << 8)
 
